homework3/Question3A&B.py

```diff
@@ -0,0 +1,80 @@
+urls = {
+    "base": "https://codefirstgirls.com/",
+    "courses": "https://codefirstgirls.com/courses",
+    "opportunities": "https://codefirstgirls.com/opportunities",
+    "cfgdegree": "https://codefirstgirls.com/courses/cfgdegree/",
+    "ambassadors": "https://codefirstgirls.com/opportunities/ambassadors/"
+}
+
+current_url = urls["base"]
+
+while True:
+    print("You are currently on the URL:", current_url)
+
+    if current_url == urls["base"]:
+        print("Where are you clicking?")
+        print("Options: Courses, Opportunities")
+        choice = input("Enter your choice: ")
+
+        if choice == "Courses":
+            current_url = urls["courses"]
+        elif choice == "Opportunities":
+            current_url = urls["opportunities"]
+        elif choice != "Courses" or "Opportunities":
+            print("Invalid choice")
+            pass
+    elif current_url == urls["courses"]:
+        print("Where are you clicking?")
+        print("Options: CFGDegree, Back")
+        choice = input("Enter your choice: ")
+
+        if choice == "CFGDegree":
+            current_url = urls["cfgdegree"]
+        elif choice == "Back":
+            current_url = urls["base"]
+        elif choice != "CFGDegree" or "Back":
+            print("Invalid Choice")
+            pass
+    elif current_url == urls["cfgdegree"]:
+        print("Where are you clicking?")
+        print("Options: Back")
+        choice = input("Enter your choice: ")
+
+        if choice == "Back":
+            current_url = urls["courses"]
+        elif choice != "Back":
+            print("Invalid choice")
+            pass
+    elif current_url == urls["opportunities"]:
+        print("Where are you clicking?")
+        print("Options: Ambassadors, Back")
+        choice = input("Enter your choice: ")
+
+        if choice == "Ambassadors":
+            current_url = urls["ambassadors"]
+        elif choice == "Back":
+            current_url = urls["base"]
+        elif choice != "Ambassadors" or "Back":
+            print("Invalid Choice")
+            pass
+    else:
+        print("Invalid URL.")
+        break
+
+
```

Comment:
Marks (12/15) - Rather than using multiple 'if-else' statements, you can enhance your program's efficiency by utilizing a dictionary. This dictionary can hold URLs as keys and their corresponding options as values. Additionally, you can manage the 'current_url' using a list data structure, which allows you to efficiently add URLs using 'append' and remove them using 'pop'. This approach improves efficiency and eliminates the need for multiple 'if-else' conditions.